### PR TITLE
Scala 3.0.0-M3 support

### PR DESCRIPTION
**Pull request by vigoo**
Solves #283

### CODE REVIEW

**Comment by vigoo** [comment]
Had to get rid of the optional dependencies for Scala 3 as it crashes the compiler. Basically a simple `implicitly[Monad[UIO]]` crashes it the `Optional` dependencies are not on the classpath. On the long term we can find alternative ways to do this or split to multiple libs, for now I think it is good to have a published Scala 3 cats interop even if it always pulls in CE and FS2

File: project/BuildHelper.scala [hunk 1]
```diff
@@ -71,25 +83,41 @@ object BuildHelper {
...
     scalaVersion in ThisBuild := crossScalaVersions.value.head,
     scalacOptions := stdOptions ++ extraOptions(scalaVersion.value),
-    libraryDependencies ++= compileOnlyDeps ++ testDeps ++ Seq(
-      compilerPlugin("org.typelevel"   % "kind-projector"  % "0.11.0") cross CrossVersion.full,
-      compilerPlugin("com.github.ghik" % "silencer-plugin" % "1.7.1") cross CrossVersion.full
-    ),
+    libraryDependencies ++= testDeps ++ {
+      if (isDotty.value)
+        Seq(
+          ("com.github.ghik" % s"silencer-lib_$Scala213" % silencerVersion % Provided)
```
Comment (neko-kai) [inline project/BuildHelper.scala:92]: We may get rid of silencer now with Scala `2.12.13` - it fully supports `@nowarn` annotation

**Comment by neko-kai** [comment]
> Basically a simple implicitly[Monad[UIO]] crashes it the Optional dependencies are not on the classpath.

That's interesting, the last time I checked dotty support for optional typeclass instances, it was _more_ permissive than Scala 2, admittedly last time was 0.27 version, not 3.0...

**Comment by vigoo** [comment]
> > Basically a simple implicitly[Monad[UIO]] crashes it the Optional dependencies are not on the classpath.
> 
> That's interesting, the last time I checked dotty support for optional typeclass instances, it was _more_ permissive than Scala 2, admittedly last time was 0.27 version, not 3.0...

Even if the `CoreSummonSpec` is reduced to

```scala
object X {
  import zio.interop.catz.core._
  implicitly[cats.Monad[zio.UIO]]
}
```

and the optional deps are kept optional, it crashes like this:

```
Error while emitting CoreSummonSpec.scala
[info] exception occurred while compiling /Users/vigoo/projects/oss/interop-cats/core-only-test/shared/src/test/scala/zio/interop/test/CoreSummonSpec.scala
java.lang.AssertionError: assertion failed while compiling /Users/vigoo/projects/oss/interop-cats/core-only-test/shared/src/test/scala/zio/interop/test/CoreSummonSpec.scala
[error] ## Exception when compiling 1 sources to /Users/vigoo/projects/oss/interop-cats/core-only-test/jvm/target/scala-3.0.0-M3/test-classes
[error] java.lang.AssertionError: assertion failed
[error] scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:11)
[error] dotty.tools.dotc.core.Types$MethodType.<init>(Types.scala:3474)
[error] dotty.tools.dotc.core.Types$CachedMethodType.<init>(Types.scala:3500)
[error] dotty.tools.dotc.core.Types$MethodTypeCompanion.apply(Types.scala:3565)
[error] dotty.tools.dotc.core.Types$MethodTypeCompanion.apply(Types.scala:3564)
[error] dotty.tools.dotc.core.Types$LambdaTypeCompanion.apply(Types.scala:3512)
[error] dotty.tools.dotc.core.TypeErasure.eraseInfo(TypeErasure.scala:552)
[error] dotty.tools.dotc.core.TypeErasure$.transformInfo(TypeErasure.scala:202)
[error] dotty.tools.dotc.transform.Erasure.transform(Erasure.scala:83)
...
```

I did not try older dotty versions

**Comment by neko-kai** [comment]
@vigoo Looks like a regression. Would you mind reporting this upstream to https://github.com/lampepfl/dotty? Yeah, it's a good call to drop Optionals for M3.

**Comment by neko-kai** [comment]
For reporting upstream you may prepare a clone of this repo with branch that reproduces it like in this report, like in this report https://github.com/lampepfl/dotty/issues/8764 no need to minimize into a completely standalone repro unless you want to

**Comment by vigoo** [comment]
@neko-kai I will report it

**Comment by neko-kai** [comment]
Thanks! 👍

**Review by neko-kai — APPROVED** [review]
🚀

**Pull request merged**